### PR TITLE
Feature skip confirmation 

### PR DESCRIPTION
**Pull request by Rambhosale**
This PR contains the continuation work from https://github.com/jesseduffield/lazygit/pull/1590
as it was closed by the PR author, made the required changes and created a new pull request

### CODE REVIEW

File: pkg/config/user_config.go [hunk 1]
```diff
@@ -11,11 +11,12 @@ type UserConfig struct {
...
-	DisableStartupPopups bool              `yaml:"disableStartupPopups"`
-	CustomCommands       []CustomCommand   `yaml:"customCommands"`
-	Services             map[string]string `yaml:"services"`
-	NotARepository       string            `yaml:"notARepository"`
+	OS                           OSConfig          `yaml:"os,omitempty"`
+	DisableStartupPopups         bool              `yaml:"disableStartupPopups"`
+	CustomCommands               []CustomCommand   `yaml:"customCommands"`
+	Services                     map[string]string `yaml:"services"`
+	NotARepository               string            `yaml:"notARepository"`
+	PromptToReturnFromSubprocess bool              `yaml:"PromptToReturnFromSubprocess"`
```
Comment (jesseduffield) [inline pkg/config/user_config.go:19]:
```suggestion
	PromptToReturnFromSubprocess bool              `yaml:"promptToReturnFromSubprocess"`
```

File: pkg/config/user_config.go [hunk 2]
```diff
@@ -533,10 +534,11 @@ func GetDefaultConfig() *UserConfig {
...
-		DisableStartupPopups: false,
-		CustomCommands:       []CustomCommand(nil),
-		Services:             map[string]string(nil),
-		NotARepository:       "prompt",
+		OS:                           GetPlatformDefaultConfig(),
+		DisableStartupPopups:         false,
+		CustomCommands:               []CustomCommand(nil),
+		Services:                     map[string]string(nil),
+		NotARepository:               "prompt",
+		PromptToReturnFromSubprocess: true,
```
Comment (jesseduffield) [inline pkg/config/user_config.go:542]: Can we specify this default value in our config doc at docs/Config.md?

File: pkg/i18n/english.go [hunk 3]
```diff
@@ -842,6 +843,7 @@ func EnglishTranslationSet() TranslationSet {
 		LcStashStagedChanges:                "stash staged changes",
 		LcStashOptions:                      "Stash options",
 		NotARepository:                      "Error: must be run inside a git repository",
+		PromptToReturnFromSubprocess:        "Display confirmation when subprocess terminates",
```
Comment (jesseduffield) [inline pkg/i18n/english.go:846]: It doesn't look like this is used. If so, we should delete it

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Thanks for making this :) I've left some comments

File: pkg/config/user_config.go (hunk 2 above)
Comment (Rambhosale) [reply pkg/config/user_config.go:542]: updated config file with promptToReturnFromSubprocess default value and short desc

File: pkg/i18n/english.go (hunk 3 above)
Comment (jesseduffield) [reply pkg/i18n/english.go:846]: I've found that typically when people want to add some string that needs to be i18'd they don't get this file first: they just assume that no matching string exists and add a new one themselves. So I'd consider this dead code and reckon we should remove it.

**Comment by jesseduffield** [comment]
I took the liberty of fixing an issue in the code and squashing the commits down. I'll merge after these tests pass. Thanks for making this :)

**Pull request merged**